File: Lib/Telerik/Examples/QuickStart/TreeView/ManipulateData/Editing/PropertyHelper.vb
Imports System.Text
Imports System.Reflection
Imports Telerik.WinControls

Namespace Telerik.Examples.WinControls.TreeView.ManipulateData.Editing
	Public Class PropertyHelper
		Private sourceObject As Object
		Private propertyInfo As PropertyInfo

		Public Sub New(ByVal sourceObject As Object, ByVal propertyName As String)
			Me.sourceObject = sourceObject
			propertyInfo = sourceObject.GetType().GetProperty(propertyName)
		End Sub

		Public Property Value() As Object
			Get
				Return propertyInfo.GetValue(sourceObject, New Object() { })
			End Get
			Set(ByVal value As Object)
				propertyInfo.SetValue(sourceObject, value, New Object() { })
			End Set
		End Property

		Public ReadOnly Property ValueType() As Type
			Get
				Return propertyInfo.PropertyType
			End Get
		End Property

		Public Overrides Function ToString() As String
			Dim value As Object = Me.Value
			If value IsNot Nothing Then
				If value.GetType() Is GetType(Color) Then
					Return (CType(value, Color)).Name
				ElseIf value.GetType() Is GetType(RoundRectShape) Then
					Dim shape As RoundRectShape = CType(value, RoundRectShape)
					Return shape.Radius.ToString()
				End If
				Return value.ToString()
			End If
			Return ""
		End Function
	End Class
End Namespace

File: Lib/Telerik/Examples/QuickStart/TreeView/ManipulateData/Editing/PropertyHelper.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Reflection;
using System.Drawing;
using Telerik.WinControls;

namespace Telerik.Examples.WinControls.TreeView.ManipulateData.Editing
{
    public class PropertyHelper
    {
        object sourceObject;
        PropertyInfo propertyInfo;

        public PropertyHelper(object sourceObject, string propertyName)
        {
            this.sourceObject = sourceObject;
            propertyInfo = sourceObject.GetType().GetProperty(propertyName);
        }

        public object Value
        {
            get
            {
                return propertyInfo.GetValue(sourceObject, new object[] { });
            }
            set
            {
                propertyInfo.SetValue(sourceObject, value, new object[] { });
            }
        }

        public Type ValueType
        {
            get { return propertyInfo.PropertyType; }
        }

        public override string ToString()
        {
            object value = Value;
            if (value != null)
            {
                if (value.GetType() == typeof(Color))
                {
                    return ((Color)value).Name;
                }
                else if (value.GetType() == typeof(RoundRectShape))
                {
                    RoundRectShape shape = (RoundRectShape)value;
                    return shape.Radius.ToString();
                }
                return value.ToString();
            }
            return "";
        }
    }
}
